Formula/youplot.rb

```diff
@@ -0,0 +1,32 @@
+class Youplot < Formula
+  desc "Command-line tool that draw plots on the terminal"
+  homepage "https://github.com/red-data-tools/YouPlot/"
+  url "https://github.com/red-data-tools/YouPlot/archive/v0.4.5.tar.gz"
+  sha256 "aa7339139bc4ea9aa0b2279e4e8052fde673a60ad47e87d50fde06626dc2b3c3"
+  license "MIT"
+
+  uses_from_macos "ruby"
+
+  resource "unicode_plot" do
+    url "https://rubygems.org/downloads/unicode_plot-0.0.5.gem"
+    sha256 "91ce6237bca67a3b969655accef91024c78ec6aad470fcddeb29b81f7f78f73b"
+  end
+
+  resource "enumerable-statistics" do
+    url "https://rubygems.org/downloads/enumerable-statistics-2.0.7.gem"
+    sha256 "eeb84581376305327b31465e7b088146ea7909d19eb637d5677e51f099759636"
+  end
+
+  def install
+    ENV["GEM_HOME"] = libexec
+    system "gem", "build", "#{name}.gemspec"
+    system "gem", "install", "#{name}-#{version}.gem"
+    bin.install libexec/"bin/#{name}", libexec/"bin/uplot"
+    bin.env_script_all_files(libexec/"bin", GEM_HOME: ENV["GEM_HOME"])
+  end
+
+  test do
+    system bin/"youplot --version"
+    system bin/"uplot --version"
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
My apologies. I added the test.

But audit now points out trailing whitespace. Any ideas?

```
youplot:
  * 34: col 1: `EOS` is not aligned with `(testpath/"test.csv").write <<~EOS` or beginning of method definition.
  * 37: col 14: Trailing whitespace detected.
  * 38: col 15: Trailing whitespace detected.
  * 39: col 16: Trailing whitespace detected.
  * 40: col 17: Trailing whitespace detected.
  * 41: col 19: Trailing whitespace detected.
```